File: src/main/java/com/gestiontache/model/Priority.java
package com.gestiontache.model;

/** Priority level of a task, from most to least urgent. */
public enum Priority {
    HAUTE("Haute"),
    MOYENNE("Moyenne"),
    BASSE("Basse");

    private final String label;

    Priority(String label) {
        this.label = label;
    }

    @Override
    public String toString() {
        return label;
    }
}

File: dotnet-winui3/GestionTaches/Models/Priority.cs
namespace GestionTaches.Models;

/// <summary>Priority level of a task, from most to least urgent.</summary>
public enum Priority
{
    HAUTE,
    MOYENNE,
    BASSE,
}

public static class PriorityExtensions
{
    /// <summary>French display label, matching the Java client's Priority.toString().</summary>
    public static string ToDisplayLabel(this Priority priority) => priority switch
    {
        Priority.HAUTE => "Haute",
        Priority.MOYENNE => "Moyenne",
        Priority.BASSE => "Basse",
        _ => priority.ToString(),
    };
}
